query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java

```diff
@@ -402,6 +402,11 @@ private Object extractKey(Object storedKey) {
    }
 
    CompletableFuture<?> processChange(InvocationContext ctx, FlagAffectedCommand command, Object storedKey, Object storedOldValue, Object storedNewValue) {
+      if (searchMapping.isRestarting()) {
+         log.mappingIsRestarting();
+         return CompletableFutures.completedNull();
+      }
```

Comment:
See... what if the restart is requested *just after this code*?

Reply:
it is possible, right! what we're trying to do is to just alter the user with a log message that is very likely to be produced, we don't need the 100% I think